content/en/blog/2023/end-user-discussions-03.md

```diff
@@ -0,0 +1,288 @@
+distro.
+
+#### 4 - Rate limiting on receivers
+
+**Q:** Are there any plans for enabling rate limiting and circuit breaks on
+receivers? Imagine having lots of clients sending telemetry to the same set of
+OTel collectors.
+
+**More context**: How do I rate-limit in a situation where I have collectors for
+traces, metrics, and logs, and I’m receiving traffic from more than 100
+individual apps? If I have even one customer who is generating heavy traffic, it
+might impact the overall health of my collectors.
+
+**A:** Use a reverse proxy. Something to note is that once the data is inside
+the collector, the data is already being deserialized, and you’ve already
+started firehosing the collector, so it’s a bit late to rate limit at that
+point. One approach might be to add additional headers when you configure your
+SDKs that contain the additional information, which would help with load
+balancing.
+
+#### 5 - Connectors
+
+**Q:** What is a connector?
+
+**A:** A connector is a collector component that consumes telemetry signals as
+an exporter in one pipeline, and emits it as a receiver in another pipeline.
+[Read more here](https://o11y.news/2023-03-13/#opentelemetry-connectors).
+
+#### 6 - Definitions of upstream, downstream, and distro
+
+**Q:** What is upstream? Downstream? Distro?
+
+**A:** The terms "upstream" and "downstream" refer to how services or components
+in a system are connected to each other. Check out
+[this article](https://reflectoring.io/upstream-downstream/) for more
+information as it applies to different situations in software.
+
+The term "distro" is short for distribution. Check out
+[this documentation](https://opentelemetry.io/ecosystem/vendors/) to see which
+vendors provide distros.
+
+### SAMPLING
+
+#### 1 - Tail sampling
+
+**Q:** What are the perceived downsides of tail sampling, for example, on all
+HTTP requests that have errors or long latencies, instead of just relying on
+head-based sampling? Are there best practices around trace sampling? Tail
+sampling can get very expensive.
+
+**A:** Generally, head sampling is not recommended, as you aren’t going to be
+able to do 100% of what you want to do with it, but it is true that tail
+sampling is expensive. The reason why sampling is such a complicated discussion
+is that there really isn’t a universal answer; furthermore, it also depends on
+what kind of features are offered by your data analysis tool. For example, do
+you have a data ingest or storage cost? If you have ingest cost, you’ll want to
+sample before the data gets ingested; if it’s storage cost, you’ll have to
+delete a lot of the data, so it depends on the tradeoffs.
+
+One thing to consider is that you can use tail sampling on attributes, such as
+if there’s an error on a span, but it does require more memory. Suggested
+further exploration:
+
+- [Column data store for OpenTelemetry](https://github.com/open-telemetry/oteps/pull/171)
+- [OpAMP](https://opentelemetry.io/blog/2022/opamp/)
```

Comment:
Drop `https://opentelemetry.io` from every link in this page that refers to a website local path:

```suggestion
- [OpAMP](/blog/2022/opamp/)
```

Reply:
> Drop `https://opentelemetry.io` from _**every link**_ in this page that refers to a website local path

Marking as unresolved: there are other links that need to have the `https://opentelemetry.io` dropped. If you'd rather I make that change, let me know.

Reply:
Oh whoops! I'll go through the links.

Reply:
Ok I think I got them all.